### PR TITLE
Add YearDate answer type

### PR DESCRIPTION
**Pull request by dwyeradam**
New YearDate answer type.
Min/Max and Range offsets can only specify years.

### CODE REVIEW

File: tests/test_schema_validation.py [hunk 1]
```diff
@@ -149,6 +149,18 @@ def test_invalid_mm_yyyy_date_range_period(self):
...
+        json_to_validate = self.open_and_load_schema_file(file)
+
+        errors = self.validator.validate_schema(json_to_validate)
+
+        self.assertEqual(len(errors), 1)
+        self.assertEqual(errors[0]['message'], 'Schema Integrity Error. Days/Months can not be used in period_limit'
+                                               ' for yyyy date range for date-range-question')
+
+
     def test_invalid_single_date_period(self):
```
Comment (MebinAbraham) [inline tests/test_schema_validation.py]: Consistency wise. Too many blank lines.

File: app/validation/validator.py [hunk 2]
```diff
@@ -240,6 +240,14 @@ def validate_date_range(self, question):
...
                                                   .format(question['id'])))
 
+            if (first_answer_type == 'YearDate'
+                    and ('minimum' in period_limits and ('days' in period_limits['minimum']
+                                                         or 'months' in period_limits['minimum'])
+                         or 'maximum' in period_limits and ('days' in period_limits['maximum']
+                                                            or 'months' in period_limits['maximum']))):
+                errors.append(self._error_message('Days/Months can not be used in period_limit for yyyy date range'
+                                                  ' for {}'.format(question['id'])))
+
```
Comment (MebinAbraham) [inline app/validation/validator.py]: Also `R:243,16: Too many boolean expressions in if statement (7/5) (too-many-boolean-expressions)`

Comment (dwyeradam) [reply app/validation/validator.py]: I can split this into two separate statements. One for Days and one for Months but that looks messy and could return twice as many errors as necessary. I'll see if I can reduce the boolean expressions but keep it simple to understand. I didn't want a complex "If" that's actually harder to understand but has less expressions.

Comment (nealedj) [reply app/validation/validator.py]: Possible suggestion:

```
if first_answer_type == 'YearDate':
    has_min_limit = 'days' in period_limits['minimum'] or 'months' in period_limits['minimum']
    has_max_limit = 'days' in period_limits['maximum'] or 'months' in period_limits['maximum']
    if has_min_limit or has_max_limit:
        errors.append(self._error_message('Days/Months can not be used in period_limit for yyyy date range'
                                        ' for {}'.format(question['id'])))
```

Comment (dwyeradam) [reply app/validation/validator.py]: As above but:

Comment (dwyeradam) [reply app/validation/validator.py]: This can actually then go above and be used for mm-yyyy type as well.

Comment (nealedj) [reply app/validation/validator.py]: Nice 👏

Comment (shahi645) [reply app/validation/validator.py]: Bravo 👍

File: tests/test_schema_validation.py (hunk 1 above)
Comment (dwyeradam) [reply tests/test_schema_validation.py]: So petty but done.

**Pull request merged**